frontend/src/views/supp-allowances/SupplementarySubmitView.vue

```diff
@@ -1,3 +1,308 @@
+    },
+    isSupportComplete() {
+      const model = this.getModel(SUPPLEMENTARY_TYPES.SUPPORT)
+      if (!model?.supportFundingModel.includes(this.SUPPORT_CHECKBOX_LABELS.find((item) => item.label === 'Other').value)) {
+        return true
+      }
+      return model?.supportFundingModel.includes(this.SUPPORT_CHECKBOX_LABELS.find((item) => item.label === 'Other').value) && !isEmpty(model.supportOtherDescription)
+    },
+    isTransportComplete() {
+      const models = this.getTransportModels()
+      return models.every((el) => el.VIN && el.estimatedMileage && el.odometer && el.uploadedDocuments?.length > 0)
+    },
+    isApplicationComplete() {
+      return this.isIndigenousComplete && this.isSupportComplete && this.isTransportComplete //&&checkbox clicked
+    },
+    readonly() {
+      return !this.isApplicationComplete || this.processing || this.loading
+    },
+  },
+  watch: {
+    back: {
+      async handler() {
+        await this.saveApplication()
+        this.$router.push({ name: 'supp-allowances-form' })
+      },
+    },
+    cancel: {
+      handler() {
+        this.toggleCancelDialog()
+      },
+    },
+    save: {
+      async handler() {
+        await this.saveApplication(true)
+      },
+    },
+    submit: {
+      async handler() {
+        await this.submitApplication()
+      },
+    },
+  },
+  async created() {
+    this.SUPPLEMENTARY_TYPES = SUPPLEMENTARY_TYPES
+    this.SUPPORT_CHECKBOX_LABELS = SUPPORT_CHECKBOX_LABELS
+    this.INDIG_CHECKBOX_LABELS = INDIG_CHECKBOX_LABELS
+    this.PANELS = [
+      {
+        title: 'Indigenous Programming Allowance',
+        id: 'indigenous',
+        supplementaryType: this.SUPPLEMENTARY_TYPES.INDIGENOUS,
+      },
+      {
+        title: 'Support Needs Programming Allowance',
+        id: 'support-needs',
+        supplementaryType: this.SUPPLEMENTARY_TYPES.SUPPORT,
+      },
+      {
+        title: 'Transportation Allowance',
+        id: 'transportation',
+        supplementaryType: this.SUPPLEMENTARY_TYPES.TRANSPORT,
+      },
+    ]
+    this.panel = this.allPanelIDs
+    await this.loadData()
+  },
+  methods: {
+    isEmpty,
+    togglePanel() {
+      this.panel = isEmpty(this.panel) ? this.allPanelIDs : []
+    },
+    async loadData() {
+      try {
+        this.loading = true
+        this.$emit('process', true)
+        //this page should specifiy to load only those applications in "draft" status - as there will be
+        //scenarios where some applications have been submitted, but the user will want to come back and fill in others.
+        this.models = await ApplicationService.getSupplementaryApplications(this.$route.params.applicationGuid)
+
+        for (const el of this.models) {
+          const found = this.PANELS.find((panel) => panel.supplementaryType == el.supplementaryType)
```

Comment:
Optional: could refactor some areas of concern out of loadData for readable/maintainable... i.e. loadDocuments(model) and setPanelId(model)